subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/api/ConfigurationRoleUsageIntegrationTest.groovy

```diff
@@ -495,20 +507,90 @@ class ConfigurationRoleUsageIntegrationTest extends AbstractIntegrationSpec {
     }
     // endregion Role-Based Configurations
 
-    // region Logging
-    def "changing usage is logged as deprecation if requested for #usageName"() {
-        given:
+    // region Warnings
+    def "changing usage for configuration #configuration produces warnings"() {
+        given: "a buildscript which attempts to change a configuration's usage"
+        buildFile << """
+            plugins {
+                id 'java-library'
+            }
+        
+            configurations {
+                $configuration {
+                    canBeResolved = !canBeResolved
+                }
+            }
+        """
+
+        expect: "the build succeeds and a deprecation warning is logged"
+        expectResolvableChanging(":$configuration", true)
+        succeeds 'help'
+
+        where: "a non-exhaustive list of configurations is tested"
+        configuration << ['api', 'implementation', 'compileOnly', 'runtimeOnly', 'archives']
+    }
+
+    def "setting consumable = false is permitted without warning for special cases to support Kotlin plugin (can change #configuration usage without warning = #allowed)"() {
```

Comment:
Given that we're specifically calling out kotlin here I think it would be a good idea to reference which code in the kotlin plugin is causing the behavior we're working around.

Reply:
Good idea.